de.fhdo.lemma.operationdsl/src/de/fhdo/lemma/operationdsl/validation/OperationDslValidator.xtend

```diff
@@ -749,4 +749,50 @@ class OperationDslValidator extends AbstractOperationDslValidator {
             ]
         }
     }
+
+    /**
+     * Check if a used by dependency is not also included in a depends on dependency
+     */
+    @Check
+    def checkDuplicatedOperationNodeDependenciesForUsedByNodes(OperationNode node) {
+        node.dependsOnNodes.forEach[dependsOnNode |
+            node.usedByNodes.forEach[usedByNode, index |
+                if (usedByNode.node == dependsOnNode.node) {
+                    error('''«node.name» may only have a dependsOn or usedBy dependency ''' +
```

Comment:
I suggest to clarify and shorten the error message, so that an instance would read something like
```"MongoDB may only depend on or being used by the same Eureka node.```, i.e., the template string would be ```«node.name» may only depend on or being used by the same «usedByNode.node.name» node.```